src/components/MonthRangeInput/MonthRangeInput.jsx

```diff
@@ -0,0 +1,73 @@
+import { format, endOfToday, addMonths } from 'date-fns';
+import { useSearch } from '@peripleo/peripleo';
+
+// Shorthand
+const fmt = date => format(date, 'yyyy-MM');
+
+// TODO lookup min date from graph!
+const minDate = new Date(1850, 0, 1);
+const maxDate = endOfToday();
+
+export const MonthRangeInput = (/*{ dateRange, min, max, onChange, onClick } */) => {
+
+  const { search, setFilter } = useSearch();
+
+  const [ from, to ] =
+    search.args.filters?.find(f => f.name === 'daterange')?.range || [ minDate, maxDate ];
+
+  const increment = (date, inc) => () => {
+    const updated = date === from ? 
+      [ addMonths(date, inc), to ] : [ from, addMonths(date, inc) ];
+
+    setFilter({ name: 'daterange', range: updated });
+  }
+
+  const onChangeDate = date => evt => {
+    const { value } = evt.target;
+
+    const updated = date === from ?
+      [ new Date(`${value}-01T00:00:00`), to ] :
+      [ from, new Date(`${value}-01T00:00:00`) ];
+
+    setFilter({ name: 'daterange', range: updated });
+  }
```

Comment:
Do these methods get re-instantiated on every re-render? If so, can we pull them out of the functional component body?

Reply:
You mean increment and onChangeDate? Yes, they get re-instantiated on every render; and re-render should happen every time the search changes (i.e. not extremely often, only on user action). Given that the methods are small, I wonder if it's worth optimizing this. Thoughts?

Reply:
That makes sense to me.